### PR TITLE
REPO-1754 creating JUnit for d:period map test

### PR DESCRIPTION
**Pull request by anechifor**
Added a new class for testing the conversion of a Map to Period.class

### CODE REVIEW

File: src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+{
+
+    public void testMapToPeriod() throws ParseException
+    {
+        Map<String, String> periodArgs = new HashMap<String, String>();
+        periodArgs.put("periodType", "day");
+        Period period = new Period(periodArgs);
+        assertNotNull(period.getPeriodType());
+        assertNull(period.getExpression());
+        assertEquals(new Period(periodArgs), DefaultTypeConverter.INSTANCE.convert(Period.class, periodArgs));
```
Comment (mattward) [inline src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java]: This class should just be testing the Period class (in particular the constructor that takes a `Map`). We don't need to test the `DefaultTypeConverter` here (and should not). This should be checking that after constructing a `Period` using a `Map`, that the state of the Period object is as we expect, e.g. `assertEquals("day", period.getPeriodType)`. The assertNotNull and assertNull are correctly checking the state of the Period.

File: src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        periodArgs.put("expression", "0");
+        period = new Period(periodArgs);
+        assertNotNull(period.getPeriodType());
+        assertNotNull(period.getExpression());
+        assertEquals(new Period(periodArgs), DefaultTypeConverter.INSTANCE.convert(Period.class, periodArgs));
+
+        try
```
Comment (mattward) [inline src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java]: Good test case. These various cases could be split into separate testXXX methods - which would provide more isolation and allow better reporting when something breaks. It's not crucial though.

File: src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+            period = new Period(periodArgs);
+            fail("the source map must not be null");
+        }
+        catch (IllegalArgumentException e)
+        {
+            // Check the exception is source.
+            assertTrue(e.getMessage().contains("Cannot create Period."));
+        }
+
+        try
```
Comment (mattward) [inline src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java:71]: Good test case

File: src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java (hunk 1 above)
Comment (anechifor) [reply src/test/java/org/alfresco/service/cmr/repository/PeriodTest.java]: ok I modify .

**Pull request merged**